src/Generator.cpp

```diff
@@ -176,6 +176,20 @@ T parse_scalar(const std::string &value) {
     return t;
 }
 
+Func make_param_func(const Parameter &p, const std::string &name) {
+    internal_assert(p.is_buffer());
```

Comment:
(Since get_buffer() is marked as "only relevant when jitting" it might be useful to have it assert-fail if called this way?)